pyomo/util/tests/test_latex_printer.py

```diff
@@ -0,0 +1,349 @@
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import pyomo.common.unittest as unittest
+from pyomo.util.latex_printer import latex_printer
+import pyomo.environ as pe
+
+
+def generate_model():
+    import pyomo.environ as pe
+    from pyomo.core.expr import Expr_if
+    from pyomo.core.base import ExternalFunction
+
+    m = pe.ConcreteModel(name='basicFormulation')
+    m.x = pe.Var()
+    m.y = pe.Var()
+    m.z = pe.Var()
+    m.objective_1 = pe.Objective(expr=m.x + m.y + m.z)
+    m.constraint_1 = pe.Constraint(
+        expr=m.x**2 + m.y**-2.0 - m.x * m.y * m.z + 1 == 2.0
+    )
+    m.constraint_2 = pe.Constraint(expr=abs(m.x / m.z**-2) * (m.x + m.y) <= 2.0)
+    m.constraint_3 = pe.Constraint(expr=pe.sqrt(m.x / m.z**-2) <= 2.0)
+    m.constraint_4 = pe.Constraint(expr=(1, m.x, 2))
+    m.constraint_5 = pe.Constraint(expr=Expr_if(m.x <= 1.0, m.z, m.y) <= 1.0)
+
+    def blackbox(a, b):
+        return sin(a - b)
+
+    m.bb = ExternalFunction(blackbox)
+    m.constraint_6 = pe.Constraint(expr=m.x + m.bb(m.x, m.y) == 2)
+
+    m.I = pe.Set(initialize=[1, 2, 3, 4, 5])
+    m.J = pe.Set(initialize=[1, 2, 3])
+    m.K = pe.Set(initialize=[1, 3, 5])
+    m.u = pe.Var(m.I * m.I)
+    m.v = pe.Var(m.I)
+    m.w = pe.Var(m.J)
+    m.p = pe.Var(m.K)
+
+    m.express = pe.Expression(expr=m.x**2 + m.y**2)
+
+    def ruleMaker(m, j):
+        return (m.x + m.y) * sum(m.v[i] + m.u[i, j] ** 2 for i in m.I) <= 0
+
+    m.constraint_7 = pe.Constraint(m.I, rule=ruleMaker)
+
+    def ruleMaker(m):
+        return sum(m.p[k] for k in m.K) == 1
+
+    m.constraint_8 = pe.Constraint(rule=ruleMaker)
+
+    def ruleMaker(m):
+        return (m.x + m.y) * sum(m.w[j] for j in m.J)
+
+    m.objective_2 = pe.Objective(rule=ruleMaker)
+
+    m.objective_3 = pe.Objective(expr=m.x + m.y + m.z, sense=-1)
+
+    return m
+
+
+def generate_simple_model():
+    import pyomo.environ as pe
+
+    m = pe.ConcreteModel(name='basicFormulation')
+    m.x = pe.Var()
+    m.y = pe.Var()
+    m.objective_1 = pe.Objective(expr=m.x + m.y)
+    m.constraint_1 = pe.Constraint(expr=m.x**2 + m.y**2.0 <= 1.0)
+    m.constraint_2 = pe.Constraint(expr=m.x >= 0.0)
+
+    m.I = pe.Set(initialize=[1, 2, 3, 4, 5])
+    m.J = pe.Set(initialize=[1, 2, 3])
+    m.K = pe.Set(initialize=[1, 3, 5])
+    m.u = pe.Var(m.I * m.I)
+    m.v = pe.Var(m.I)
+    m.w = pe.Var(m.J)
+    m.p = pe.Var(m.K)
+
+    def ruleMaker(m, j):
+        return (m.x + m.y) * sum(m.v[i] + m.u[i, j] ** 2 for i in m.I) <= 0
+
+    m.constraint_7 = pe.Constraint(m.I, rule=ruleMaker)
+
+    def ruleMaker(m):
+        return sum(m.p[k] for k in m.K) == 1
+
+    m.constraint_8 = pe.Constraint(rule=ruleMaker)
+
+    return m
+
+
+def generate_simple_model_2():
+    import pyomo.environ as pe
```

Comment:
No need to re-import

Reply:
I prefer to have the generate model function be copy-paste-able and believe this import is of minimal overhead